bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java

```diff
@@ -553,7 +553,7 @@ private void connectToDatabase() {
         } catch (Exception e) {
             logger.error(
                     "mySQL: Failed connecting to the SQL database using: driverClass={}, url={}, user={}, password={}",
-                    driverClass, url, user, password, e);
+                    driverClass, url, user, password.replaceAll(".", "*"), e);
```

Comment:
You are still exposing some information about the password here (its length), which should also be avoided, see the comment at https://github.com/eclipse/smarthome/issues/4583#issuecomment-349715045.

So I'd suggest to either put 8 asterisks or maybe simply remove the password part completely here as it anyhow is not helpful.

Reply:
@kaikreuzer, I like that, shall I also remove it then in all other places in this repo.

Should I also apply it for trace logging? or is that configurable and thus okay.

Reply:
Yes, can be removed from other places as well from the logging.
I'd also remove it from trace statements.